File: CS/XPO_MultiResource_Example/XPResource.cs
using DevExpress.Xpo;
using System;
using System.Drawing;

namespace XPO_MultiResource_Example {
    #region #xpresource
    // XP object
    public class XPResource : XPObject {
        public XPResource(Session session) : base(session) { }
        public int ResId;
        [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
        public string Name;              // Resource.Caption
        public Int32 Color;
        public Image Image;

        [Association()]
        public XPCollection<XPAppointment> Appointments {
            get {
                return GetCollection<XPAppointment>("Appointments");
            }
        }
    }
    #endregion #xpresource
}

File: VB/XPO_MultiResource_Example/XPResource.vb
Imports DevExpress.Xpo
Imports System
Imports System.Drawing

Namespace XPO_MultiResource_Example
	#Region "#xpresource"
	' XP object
	Public Class XPResource
		Inherits XPObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public ResId As Integer
		<Size(SizeAttribute.Unlimited)>
		Public Name As String ' Resource.Caption -  !!! To set the Memo field type.
		Public Color As Int32
		Public Image As Image

		<Association()>
		Public ReadOnly Property Appointments() As XPCollection(Of XPAppointment)
			Get
				Return GetCollection(Of XPAppointment)("Appointments")
			End Get
		End Property
	End Class
	#End Region ' #xpresource
End Namespace